### ISSUE TITLE
Crash when trying to shared downloaded PDF

### ISSUE THREAD

**Issue by badboy** [mozilla-mobile/firefox-ios#11777]
## Steps to reproduce

1. Go to https://fnordig.de/talks/
2. Long press the first [slides](https://fnordig.de/talks/2022/dataclub/glean-uniffi-migration.pdf) link and click "Download Link", then click "Download Now"
3. Go to the "Downloads" view (through the menu)
4. Open the just downloaded PDF
5. Open the Menu, and click "Share Page With..."

### Expected behavior

I get the iOS share view, able to send the PDF as a message, email, etc.

### Actual behavior

I get a white empty screen. The app is stuck and after some few seconds it crashes.

### Device & build information
* Device: iPhone 11, iOS 15.6.1
* Build version: 105.0 (16923) (beta from TestFlight), but also reported for 104.0 (release)

### Notes
Attachments: 

Crash log: [Client-2022-09-01-140210.ips.txt](https://github.com/mozilla-mobile/firefox-ios/files/9469605/Client-2022-09-01-140210.ips.txt)

```
  "exception" : {"codes":"0x0000000000000000, 0x0000000000000000","rawCodes":[0,0],"type":"EXC_CRASH","signal":"SIGKILL"},
  "termination" : {"code":2343432205,"flags":6,"namespace":"FRONTBOARD","reasons":["<RBSTerminateContext| domain:10 code:0x8BADF00D explanation:scene-update watchdog transgression: application<org.mozilla.ios.FirefoxBeta>:93297 exhausted real (wall clock) time allowance of 10.00 seconds","ProcessVisibility: Background","ProcessState: Running","WatchdogEvent: scene-update","WatchdogVisibility: Background","WatchdogCPUStatistics: (","\"Elapsed total CPU time (seconds): 9.040 (user 9.040, system 0.000), 14% CPU\",","\"Elapsed application CPU time (seconds): 0.233, 0% CPU\"",") reportType:CrashLog maxTerminationResistance:Interactive>"]},
```

Screenshot from when the page blanks:

<details>

![IMG_3255](https://user-images.githubusercontent.com/2129/187909979-519b81dd-bfcd-415e-a232-1d5e88f2ce4e.PNG)

</details>

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-4856)

**Comment by TwizzyIndy** [comment]
I just noticed that many people facing [this issue](https://developer.apple.com/forums/thread/703145). I've found a solution for that on [stackoverflow](https://stackoverflow.com/a/72421484/4400246). He created a `CustomShareSheet` class that extends `UIActivityViewController` and this class will dismiss the `CustomShareSheetController` when the application will resign from its active state, thus the crash won't occur anymore. I wish I could reproduce this, but I don't currently own an iOS 15.4~15.6 device.

**Comment by dnarcese** [comment]
@SimonBasca have you ever seen this crash before? I want to understand if it happens on other pdf files as well or is specific to this site.

**Comment by eimann** [comment]
Happens to me with all PDFs, not site-specific.

**Comment by TwizzyIndy** [comment]
We've got mistakenly used `tab.canonicalURL?.displayURL` as the file opening URL in L547. That let the app trigger the exception like `-[NSFileManager fileSystemRepresentationWithPath:]: nil or empty path argument`.

https://github.com/mozilla-mobile/firefox-ios/blob/c9f35c27901889254466287f93575db5c4fcc427/Client/Frontend/Browser/MainMenuActionHelper.swift#L540-L553

We should get the file `URL` like `tab.url` below. I would love to create PR for this issue. Thanks

```swift
            guard let tab = self.selectedTab,
                  let url = tab.url,
                  let presentableVC = self.menuActionDelegate as? PresentableVC
            else { return }

            self.share(fileURL: url, buttonView: self.buttonView, presentableVC: presentableVC)
        }.items
```

https://user-images.githubusercontent.com/6712795/189335794-8ea4e5f6-46fb-4af0-a77a-daf5f6928eb8.mov

**Comment by lmarceau** [comment]
@TwizzyIndy I had a look and this indeed seems to be the case. You can tag me for review when you open the PR.

**Comment by data-sync-user** [comment]
➤ Andrei Bodea commented:

Verified as fixed on the v9000 (18422) with iPhone 13 Pro (15.1.1).

I was able to share the downloaded PDF without issues with many apps.

**Comment by data-sync-user** [comment]
➤ Simion Basca commented:

Verified fix on v106.0 (19965)
